pr_toolbox/quantum/results/reckoning.py

```diff
@@ -141,67 +147,77 @@ def reckon_counts(self, counts: Counts) -> ReckoningResult:
+    def _reckon_pauli(self, frequencies: QuasiDistribution, pauli: Pauli) -> ReckoningResult:
         mask = pauli_integer_mask(pauli)
-        counts = bitmask_counts(counts, mask)
+        frequencies = bitmask_frequencies(frequencies, mask)
         coeff = (-1j) ** pauli.phase
-        expval, std_error = self._reckon_counts(counts)
+        expval, std_error = self._reckon_frequencies(frequencies)
         return ReckoningResult(coeff * expval, std_error)
 
     @abstractmethod
-    def _reckon_counts(self, counts: Counts) -> ReckoningResult:
-        shots = counts.shots() or 1  # Note: avoid division by zero errors
+    def _reckon_frequencies(self, frequencies: QuasiDistribution) -> ReckoningResult:
+        shots = frequencies.shots or 1
         expval: float = 0.0
-        for readout, freq in counts.int_outcomes().items():
+        for readout, freq in frequencies.items():
             observation = (-1) ** parity_bit(readout, even=True)
-            expval += observation * freq / shots
+            expval += observation * freq
         variance = 1 - expval**2
         std_error = sqrt(variance / shots)
         return ReckoningResult(expval, std_error)
 
     ################################################################################
     ## AUXILIARY
     ################################################################################
+
     @classmethod
-    def _validate_counts_list(cls, counts_list: Sequence[Counts] | Counts) -> tuple[Counts, ...]:
-        """Validate counts list."""
-        if isinstance(counts_list, (Counts, dict)):
-            counts_list = (counts_list,)
-        if not isinstance(counts_list, Sequence):
+    def _validate_frequencies_list(
+        cls, frequencies_list: Sequence[FrequenciesLike] | FrequenciesLike
+    ) -> tuple[QuasiDistribution, ...]:
+        """Validate frequencies."""
+        if isinstance(frequencies_list, (Counts, QuasiDistribution, dict)):
+            frequencies_list = (frequencies_list,)
+        if not isinstance(frequencies_list, Sequence):
             raise TypeError("Expected Sequence object.")
-        return tuple(cls._validate_counts(c) for c in counts_list)
+        return tuple(cls._validate_frequencies(f) for f in frequencies_list)
 
     @staticmethod
-    def _validate_counts(counts: Counts | dict) -> Counts:
-        """Validate counts."""
-        if isinstance(counts, dict):
-            counts = Counts(counts)
-        if not isinstance(counts, Counts):
-            raise TypeError("Expected Counts object.")
-        return counts
+    def _validate_frequencies(frequencies: FrequenciesLike) -> QuasiDistribution:
+        """Validate frequencies."""
+        if isinstance(frequencies, QuasiDistribution):
+            return frequencies
+        if isinstance(frequencies, (Counts, dict)):
+            frequencies = Counts(frequencies)
+            frequencies = convert_counts_to_quasi_dists(frequencies)
```

Comment:
Slightly more readable (notice the subtlety of not doing `else if`):
```suggestion
        if isinstance(frequencies, dict):
            frequencies = Counts(frequencies)
        if isinstance(frequencies, Counts):
            frequencies = convert_counts_to_quasi_dists(frequencies)
```